### ISSUE TITLE
Difference between code and paper

### ISSUE THREAD

**Issue by lizleo** [zhangxiaosong18/FreeAnchor#38]
When 

<a href="https://www.codecogs.com/eqnedit.php?latex=L_{ij}^{cls}=BCE(a_{j}^{cls},&space;b_{i}^{cls}))=-[b_{i}^{cls}\log{a_{j}^{cls}}&space;&plus;&space;(1-b_{i}^{cls})\log{(1-a_{j}^{cls})}]&space;\neq&space;BCE(a_{j}^{cls},&space;\vec{1}))," target="_blank"><img src="https://latex.codecogs.com/gif.latex?L_{ij}^{cls}=BCE(a_{j}^{cls},&space;b_{i}^{cls}))=-[b_{i}^{cls}\log{a_{j}^{cls}}&space;&plus;&space;(1-b_{i}^{cls})\log{(1-a_{j}^{cls})}]&space;\neq&space;BCE(a_{j}^{cls},&space;\vec{1}))," title="L_{ij}^{cls}=BCE(a_{j}^{cls}, b_{i}^{cls}))=-[b_{i}^{cls}\log{a_{j}^{cls}} + (1-b_{i}^{cls})\log{(1-a_{j}^{cls})}] \neq BCE(a_{j}^{cls}, \vec{1}))," /></a>

so

<a href="https://www.codecogs.com/eqnedit.php?latex=p_{ij}^{cls}=e^{-L_{ij}^{cls}}&space;\neq&space;e^{\log{a_{j}^{cls}}}," target="_blank"><img src="https://latex.codecogs.com/gif.latex?p_{ij}^{cls}=e^{-L_{ij}^{cls}}&space;\neq&space;e^{\log{a_{j}^{cls}}}," title="p_{ij}^{cls}=e^{-L_{ij}^{cls}} \neq e^{\log{a_{j}^{cls}}}," /></a>

then

<a href="https://www.codecogs.com/eqnedit.php?latex=p_{ij}^{cls}\neq&space;a_{j}^{cls}," target="_blank"><img src="https://latex.codecogs.com/gif.latex?p_{ij}^{cls}\neq&space;a_{j}^{cls}," title="p_{ij}^{cls}\neq a_{j}^{cls}," /></a>

but in your code https://github.com/zhangxiaosong18/FreeAnchor/blob/master/maskrcnn_benchmark/modeling/rpn/free_anchor_loss.py#L161,  
you just use  <a href="https://www.codecogs.com/eqnedit.php?latex=a_{j}^{cls}" target="_blank"><img src="https://latex.codecogs.com/gif.latex?a_{j}^{cls}" title="a_{j}^{cls}" /></a> (**matched_cls_prob** in your code) as  <a href="https://www.codecogs.com/eqnedit.php?latex=p_{ij}^{cls}" target="_blank"><img src="https://latex.codecogs.com/gif.latex?p_{ij}^{cls}" title="p_{ij}^{cls}" /></a>, 
that means you just ignore the other predicted classes which not matching the target class, and I think it's different with **retinanet_cls_loss** defined in https://github.com/zhangxiaosong18/FreeAnchor/blob/master/maskrcnn_benchmark/modeling/rpn/retinanet_loss.py#L142.

I try to rewrite the code calculating **matched_cls_prob** as blew:
```
labels_mul = torch.zeros([len(labels_), self.num_classes])
for i in range(len(labels_)):
    labels_mul[i, labels_[i]] = 1

labels_mul = labels_mul.unsqueeze(1).repeat(1, self.pre_anchor_topk, 1)

loss_mul_class = nn.BCELoss(reduction="none")(cls_prob_[matched], labels_mul).sum(dim=-1)
matched_cls_prob = (-loss_mul_class).exp()
```

Did I get it wrong ？ @zhangxiaosong18

**Comment by liuyang-ict** [comment]
You know if b_{i}^{cls}[k] =1, the only way it can happen is if k = i, because b_{i}^{cls} is a one-hot label.
